ogr/ogrsf_frmts/gmlutils/gmlfeatureclass.cpp

```diff
@@ -1124,3 +1123,37 @@ OGRFieldType GML_GetOGRFieldType(GMLPropertyType eType,
         eFType = OFTStringList;
     return eFType;
 }
+
+/************************************************************************/
+/*                       GML_FromOGRFieldType()                          */
+/************************************************************************/
+
+GMLPropertyType GML_FromOGRFieldType(OGRFieldType eType)
```

Comment:
you propably need to take into account both OGRFieldType and OGRFieldSubType to be able to map (OFTInteger, OFSTBool) to GMLPT_Boolean and (OFTInteger, OFSTInt16) to GMLPT_Short

Reply:
done, and test added